### PR TITLE
Tester compatible link generator

### PR DESCRIPTION
**Pull request by spagr**
Few changes in DI for succes Integration presenter testing e.g. https://github.com/mangoweb-backend/presenter-tester
Nette\Application dependency replaced with Nette\LinkGenerator
Added function MenuComponent::attached($presenter) for inject Presenter to MenuItem(s) after component attachment to presenter. This is only need for isActive() method.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -16,11 +16,15 @@
 		"issues": "https://github.com/Carrooi/Nette-Menu/issues"
 	},
 	"require": {
-		"php": ">=7.1",
+		"nette/application": "^2.4",
```
Comment (davidkudera) [inline composer.json]: Do we need all these new dependencies? And if yes, shoudn't they be in the `requireDev` section?

File: src/Carrooi/Menu/AbstractMenuItemsContainer.php [hunk 2]
```diff
@@ -27,28 +27,36 @@
...
 
+	/** @var \Nette\Application\UI\Presenter */
+	protected $presenter;
+
 	/** @var \Carrooi\Menu\IMenuItem[] */
 	private $items = [];
 
 
-	public function __construct(ILinkGenerator $linkGenerator, ITranslator $translator, IAuthorizator $authorizator, Application $application, Request $httpRequest, IMenuItemFactory $menuItemFactory)
+	public function __construct(ILinkGenerator $linkGenerator, ITranslator $translator, IAuthorizator $authorizator,  \Nette\Application\LinkGenerator $nativeLinkGenerator, Request $httpRequest, IMenuItemFactory $menuItemFactory)
```
Comment (davidkudera) [inline src/Carrooi/Menu/AbstractMenuItemsContainer.php]: Please import the namespace with `use`

File: src/Carrooi/Menu/DI/MenuExtension.php [hunk 3]
```diff
@@ -127,7 +126,7 @@ private function loadMenuConfiguration(ContainerBuilder $builder, string $menuNa
 				$linkGenerator,
 				$translator,
 				$authorizator,
-				'@'. Application::class,
+				'@'. \Nette\Application\LinkGenerator::class,
```
Comment (davidkudera) [inline src/Carrooi/Menu/DI/MenuExtension.php]: Please import the namespace with `use`

File: src/Carrooi/Menu/UI/MenuComponent.php [hunk 4]
```diff
@@ -30,6 +30,16 @@ public function __construct(MenuContainer $container, string $name)
 		$this->menuName = $name;
 	}
 
+	protected function attached($presenter)
```
Comment (davidkudera) [inline src/Carrooi/Menu/UI/MenuComponent.php]: Please add return type `: void`

File: src/Carrooi/Menu/UI/MenuComponent.php [hunk 5]
```diff
@@ -30,6 +30,16 @@ public function __construct(MenuContainer $container, string $name)
...
+	{
+		parent::attached($presenter);
```
Comment (davidkudera) [inline src/Carrooi/Menu/UI/MenuComponent.php]: Please add a new blank line after parent call

File: tests/CarrooiTests/Menu/MenuItemTest.phpt [hunk 6]
```diff
@@ -67,57 +67,29 @@ final class MenuItemTest extends TestCase
...
-			);
 		});
 
-		$item = new MenuItem($linkGenerator, $translator, $authorizator, $application, $request, $itemFactory, 'item');
-		$item->setAction('Home:');
-
-		Assert::true($item->isActive());
-	}
-
-	public function testIsActive_include(): void
```
Comment (davidkudera) [inline tests/CarrooiTests/Menu/MenuItemTest.phpt]: Same as above. We shouldn't remove the include functionality.

File: tests/CarrooiTests/TestCase.php [hunk 7]
```diff
@@ -104,6 +97,11 @@ protected function createMockHttpUrl(callable $fn = null): Url
 		return $this->createMock(Url::class, $fn);
 	}
 
+	protected function createMockNativeLinkGenerator(callable $fn = null): \Nette\Application\LinkGenerator
+	{
+		return $this->createMock(\Nette\Application\LinkGenerator::class, $fn);
```
Comment (davidkudera) [inline tests/CarrooiTests/TestCase.php]: Please import the namespace with `use`

File: tests/CarrooiTests/TestCase.php [hunk 8]
```diff
@@ -81,15 +80,9 @@ protected function createMockAuthorizator(callable $fn = null): IAuthorizator
...
+	protected function createMockPresenter(callable $fn = null): \Nette\Application\UI\Presenter
 	{
-		return $this->createMock(Application::class, $fn);
-	}
-
-
-	protected function createMockPresenter(callable $fn = null): IPresenter
-	{
-		return $this->createMock(IPresenter::class, $fn);
+		return $this->createMock(\Nette\Application\UI\Presenter::class, $fn);
```
Comment (davidkudera) [inline tests/CarrooiTests/TestCase.php]: Please import the namespace with `use`

File: tests/CarrooiTests/TestCase.php [hunk 9]
```diff
@@ -81,15 +80,9 @@ protected function createMockAuthorizator(callable $fn = null): IAuthorizator
 	}
 
 
-	protected function createMockApplication(callable $fn = null): Application
+	protected function createMockPresenter(callable $fn = null): \Nette\Application\UI\Presenter
```
Comment (davidkudera) [inline tests/CarrooiTests/TestCase.php]: Please import the namespace with `use`

File: composer.json (hunk 1 above)
Comment (spagr) [reply composer.json]: Maybe not, removing ...

**Comment by spagr** [comment]
Changes done.

**Comment by davidkudera** [comment]
Closing in favor of #32. Thank you anyway

**Pull request closed without merging**